### ISSUE TITLE
How can we send logs to logstash on TCP ports

### ISSUE THREAD

**Issue by rsingh2411** [looplab/logspout-logstash#16]
This is not actually an issue,  Correct me if i am wrong, making change in logstash.go file of udp to tcp will do the trick,
Please advise

**Comment by zot24** [comment]
I'll be interested in this feature as well as, but be able to choose between `tcp/upd` will be a better idea than change `logstast.go` file, like on a config file or from the `modules.go` file like:

```
import (
  _ "github.com/gliderlabs/logspout/transports/tcp"
  _ "github.com/looplab/logspout-logstash/transports/tcp"
)
```

or

```
import (
  _ "github.com/gliderlabs/logspout/transports/udp"
  _ "github.com/looplab/logspout-logstash/transports/udp"
)

```

**Comment by jmreicha** [comment]
+1

**Comment by rchicoli** [comment]
This should be solved with this PR
https://github.com/looplab/logspout-logstash/pull/29

**Comment by sfriedel** [comment]
Actually it doesn't need a change. You can just use `logstash+tcp://...` instead of `logstash://...` and it should work.

For it to work your logspout `modules.go` also has to include:

```
import (
  _ "github.com/gliderlabs/logspout/transports/tcp"
)
```

**Comment by rchicoli** [comment]
hey @sfriedel, it didn't work for me like you said and to be honest, it doesn't make sense either to start a transport with udp and type tcp in the protocol.

``` bash
root@9b5c57baa541:/# netstat -putan
Active Internet connections (servers and established)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp6       0      0 :::5003                 :::*                    LISTEN      -

$ docker run --rm --volume=/var/run/docker.sock:/var/run/docker.sock -e ROUTE_URIS=logstash+tcp://172.17.0.3:5003 test:1.0

# logspout v3.1-custom by gliderlabs
# adapters: raw tcp udp logstash
# options : persist:/mnt/routes
# jobs    : pump routes http[]:80
# routes  :
#   ADAPTER     ADDRESS         CONTAINERS      SOURCES OPTIONS
#   logstash+tcp172.17.0.3:5003                         map[]
```

As soon as you try to send some log information, the connection breaks:

``` bash
2016/08/09 20:12:39 logstash:write tcp 172.17.0.5:44234->172.17.0.3:5003: write: broken pipe
```

**Comment by sfriedel** [comment]
hey @rchicoli, i don't know exactly why it doesn't work for you that way. I'm currently using it with `logstash+tls://` without problems but during testing i was using `logstash+tcp://`. Did you change your logstash configuration to enable the tcp input?

> to be honest, it doesn't make sense either to start a transport with udp and type tcp in the protocol.

I'm not sure i understand completely. If you mean the `udp` that's passed to `route.AdapterTransport` then that is the [default transport](https://github.com/gliderlabs/logspout/blob/master/router/types.go#L76) that logspout will use if the url scheme doesn't specify one (so `logstash://` is actually `logstash+udp://` internally).

**Comment by rchicoli** [comment]
ok @sfriedel thx for the reply.
No idea why it didn't work for me.
Oh well, I will give another try then.

**Comment by rchicoli** [comment]
I retried with following module.go

```go
package main

import (
        _ "github.com/gliderlabs/logspout/transports/tcp"
        _ "github.com/looplab/logspout-logstash"
)
```

but it didn't work again. I would say you have used a different logspout-lostash repo..

**Comment by rchicoli** [comment]
hey @sfriedel , I tried again, somehow I got it working, like you said so.. maybe there was some old cache image on my pc, after deleting everything and rebuilding I can confirm, it does work!

**Comment by rchicoli** [comment]
this can be closed, thx to @sfriedel 
I've updated the README.

**Comment by maxekman** [comment]
Thanks for sorting this out, fixed in #29.
